### DISCUSSION TITLE
Steps to launch event listeners with postman testing?

### DISCUSSION THREAD

**Discussion by henry-ht** [imdhemy/laravel-in-app-purchases#26] [General]
Kindly list the series of more detailed steps to get the laravel event listeners up and running with purchase.

I am trying to check if the events are working but what I find when doing a post request to the default route is nothing and I find myself in a situation where I don't know what to do.

I already have all the google play settings and it works perfectly.

I have not been able to get the event to execute when making the post request, which returns to my home page

thanks

**Comment by imdhemy** [comment]
Hi, @henry-ht,

Events are triggered by the server notifications sent by (Google Play or App Store). 

**To trigger the events**:
1. Configure the server notifications of the required provider. ([RTDN](https://developer.android.com/google/play/billing/getting-ready#configure-rtdn), [ASSN](https://developer.apple.com/documentation/appstoreservernotifications))
2. Purchase a subscription from your mobile application, either by a Test account or a real purchase process.

For Google, you can send a Test Notification from your console, which will be caught by Laravel Log.

**For local development**
I'm currently working on Fakers to make the testing process easier and more productive

**Comment by henry-ht** [comment]
I told him that these configurations were already in place, I tried with a purchase from google play and it shows the purchase request and the renewal request, everything works fine. in laravel I created the listener and added it to the config file. Is there a way to test that the events you have added are running?

**Reply by imdhemy** [reply]
Oh, Okay! Events are already tested during the development of the package, but if you mean to test your **listeners**, You can dispatch the event within your test classes.

Here is a Google Example:

```php
public function test_subscription_is_deleted_if_expired(){
// Given
$data = 'valid_purchase_token';
$notification = DeveloperNotification::parse($data);
$subscriptionNotification = $notification->getSubscriptionNotification();

// When
$event = new SubscriptionExpired(new GoogleServerNotification($notification));
$listener = new HandleExpiration();
$listener->handle($event);

// Then
$this->assertDatabaseMissing('purchased_subscriptions', [
            'user_id' => $user->getId(),
        ]);
}
```

**Comment by henry-ht** [comment]
do you have a purchase json that is sent to trigger an event?

sorry for the inconvenience
